sanic/response.py

```diff
@@ -97,40 +97,39 @@ def __init__(self, body=None, status=200, headers=None,
     def output(self, version="1.1", keep_alive=False, keep_alive_timeout=None):
         # This is all returned in a kind-of funky way
         # We tried to make this as fast as possible in pure python
-        timeout_header = b''
         if keep_alive and keep_alive_timeout:
-            timeout_header = b'Keep-Alive: timeout=%d\r\n' % keep_alive_timeout
-
+            if 'Keep-Alive' not in self.headers:
+                self.headers['Keep-Alive'] = keep_alive_timeout
+        if 'Connection' not in self.headers:
+            if keep_alive:
+                self.headers['Connection'] = 'keep-alive'
+            else:
+                self.headers['Connection'] = 'close'
+        if 'Content-Length' not in self.headers:
+            self.headers['Content-Length'] = len(self.body)
+        if 'Content-Type' not in self.headers:
+            self.headers['Content-Type'] = self.content_type
         headers = b''
         if self.headers:
             for name, value in self.headers.items():
                 try:
-                    headers += (
-                        b'%b: %b\r\n' % (name.encode(), value.encode('utf-8')))
+                    headers += (b'%b: %b\r\n' % (
```

Comment:
I prefer the hanging indent here, it keeps all the logic on the same line. Same goes for the line below it as well